portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -116,6 +138,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     response.sendRedirect("/");
   }
 
+  /**
+   * Set the max number of comments that can be shown (value between 1 and 50).
+   */
+  public void setCommentMax(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
hmm, so, as you mentioned in your comment on this PR, handling the commentMax in a POST has some interesting consequences. One thing I'm thinking about is, if you had two different viewers of your website at the same time (eg: if you started up the server then sent Bootsie and I the same link), if one of us set a commentMax, the other would suddenly see that same max applied to their view of the page, since the server keeps one commentMax value which it uses to serve all requests.

There's a few ways this could be addressed, though I'm not saying you need to do any of this in this PR. If you added accounts for users on your site then when they post a commentMax you could save the max paired with the user in DatastoreService (fwiw this how some websites keep your preferences, eg: a users preference for dark-mode in gmail). Alternatively you could keep the users commentMax saved in a field in javascript, then send it up with each fetch request. But, (I think) you'd loose their preference each time the server sent a redirect response, or if they closed their browser. FWIW, I believe browser cookies are a solution to saving state on the client side between visits.

Both cookies and user accounts are probably out of scope for these walk-throughs. Are there other approaches you've thought about?

Reply:
Depending on the availability of **week-4-libraries** next week, I could certainly explore browser cookies; this could likely be important to the capstone project, after all. I didn't consider that the comment limit would refresh for all users; however, it's true that the recommended approach means that upon refresh, the comment limit would have to be reset. The only other approach I considered was the recommended one above (mostly because the others seemed out of scope as you mentioned), but it's definitely applicable and I'd love to explore that!

Reply:
As we talked about in the sync this morning, whatever path forward we settle on here can/should be done in a separate CL. I'm fine with submitting this in it's current implementation. For now, I'd suggestion prioritizing the API integrations which are scheduled for this week. And we can continue to discuss this specific topic and the alternate approaches since storing user state & preferences may well be part of the capstone depending upon what you and Eshika and Adam design.